tests/test_decorators.py

```diff
@@ -135,11 +136,50 @@ def test_cognito_login_cookie_samesite(client, cfg, access_token, token_response
         assert "SameSite=Strict" in response.headers["Set-Cookie"]
 
 
+def test_cognito_refresh_missing_token(
+    client, cfg, access_token, refresh_token_response
+):
+    with pytest.raises(
+        CognitoError, match="Refresh token is required to refresh the access token"
+    ):
+        client.get("/refresh")
+
+
+def test_cognito_refresh_callback(client, cfg, access_token, refresh_token_response):
+    with client as c:
+        with c.session_transaction() as sess:
+            # Set the refresh_token in the session
+            sess["refresh_token"] = "test_refresh_token"
+
+        # returns OK and sets the cookie
+        response = client.get("/refresh")
+        assert response.status_code == 200
+        assert response.data.decode("utf-8") == "ok"
+        assert response.headers["Set-Cookie"].startswith(
+            f"{cfg.COOKIE_NAME}={access_token}"
+        )
+
+        # removes one-time use codes from the session
+        assert "code_verifier" not in session
+        assert "nonce" not in session
+
+        # check that user claims and user info are stored in the session
+        assert "claims" in session
+        assert "user_info" in session
+        assert "refresh_token" in session
+
+
 def test_cognito_logout(client, cfg):
-    # should 302 redirect to cognito
-    response = client.get("/logout")
-    assert response.status_code == 302
-    assert response.headers["location"].startswith(cfg.logout_endpoint)
+    with client as c:
```

Comment:
Do you want to add this logic as a separate test?
Just to ensure we have coverage for both cases, i.e. standard logout flow, plus logout flow if there was a refresh token set

Reply:
Split